R/lengths_linter.R

```diff
@@ -0,0 +1,33 @@
+#' Require usage of `lengths()` where possible
+#'
+#' [lengths()] is a function that was added to base R in version 3.2.0 to
+#'   get the length of each element of a list. It is equivalent to
+#'   `sapply(x, length)`, but faster and more readable.
+#'
+#' @evalRd rd_tags("lengths_linter")
+#' @seealso [linters] for a complete list of linters available in lintr.
+#' @export
+lengths_linter <- function() {
+  loop_funs <- c("sapply", "vapply", "map_int", "map_dbl")
+  xpath <- glue::glue("
+  //SYMBOL_FUNCTION_CALL[ {xp_text_in_table(loop_funs)} ]
+  /parent::expr/parent::expr[expr[position() = 3 and SYMBOL[text() = 'length']]]
```

Comment:
Do you think it's worth to not assume positional arguments here?

Reply:
I think this is kind of a general thing in `lintr`... a few thoughts:

 - Using arguments in their positional order is good practice in general, especially for the most common/simple functions. `sapply(FUN = length, x = x)` would be 🚨 for me. Probably we should document this "philosophy" somewhere.
 - We should catch the case when the argument is named, `FUN = length`, regardless of position
 - We should always keep pipes in mind (`|>` / `%>%`)

For this particular case, I only really caught usages of `sapply(x, length)`, the others are just there because "why not". Even at that it was pretty rare.

Reply:
I have created an issue to track this : https://github.com/r-lib/lintr/issues/1573

I agree that these would be good improvements to implement at some point, but they shouldn't be the blocking factor for this PR.